implementations/rust/ockam/ockam_node/src/lib.rs

```diff
@@ -9,9 +11,71 @@
     unused_qualifications
 )]
 
+#[cfg(all(feature = "std", feature = "alloc"))]
+compile_error!(r#"Cannot compile both features "std" and "alloc""#);
+
+#[cfg(all(feature = "no_std", not(feature = "alloc")))]
+compile_error!(r#"The "no_std" feature currently requires the "alloc" feature"#);
+
+#[cfg(feature = "no_std")]
+#[macro_use]
+extern crate core;
+
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate std;
+
+#[cfg(feature = "alloc")]
+#[allow(unused_imports)]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(feature = "std")]
 #[macro_use]
 extern crate tracing;
 
+#[cfg(feature = "std")]
+pub use tokio;
+#[cfg(feature = "no_std")]
+pub use ockam_node_no_std::tokio;
```

Comment:
Not a real review but I love how you implement the embedded shim in this. I was wondering while writing the default node whether a real embedded port would even be necessary. Very happy to see that an async shim to map onto the context_m async stuff is enough 🥳